core/src/main/scala/org/apache/spark/sql/collection/ConcurrentSegmentedHashMap.scala

```diff
@@ -227,38 +236,33 @@ private[sql] class ConcurrentSegmentedHashMap[K, V, M <: SegmentMap[K, V] : Clas
             while (idx < nhashes) {
               added = seg.changeValue(keys(idx), bucketId(hashes(idx)), change, isLocal)
               if (added != null) {
-                if (added.booleanValue()) numAdded += 1
+                if (added.booleanValue()) {
+                  numAdded += 1
+                }
                 idx += 1
               } else {
                 // indicates that loop must be broken immediately
                 // need to take the latest reference of segmnet
                 // after segmnetAbort is successful
+                addNumToSize()
```

Comment:
move it to below lockedState=false just from code logic perspective in case unlock throws an exception (e.g. interrupted)